tests/unit_tests/encoder/categorical/test_binary.py

```diff
@@ -0,0 +1,145 @@
+import unittest
+import torch
+from torch import Tensor
+import numpy as np
+from lightwood.encoder.categorical.binary import (
+    BinaryEncoder,
+)
+
+from lightwood.helpers.constants import _UNCOMMON_WORD, _UNCOMMON_TOKEN
+
+
+class TestBinary(unittest.TestCase):
+    """ Test the OHE vector encoder """
+
+    def test_encode_decode_with_binary(self):
+        """
+        Test binary example case.
+        """
+        # Generate categories with None
+        data = ["apple", "apple", "orange", None, "apple", "orange"]
+
+        # Generate test category with unseen examples (banana, None)
+        test_data = ["apple", "banana", "orange", None, "apple"]
+
+        # Ground-truth answer
+        ytest = ["apple", _UNCOMMON_WORD, "orange", _UNCOMMON_WORD, "apple"]
+
+        enc = BinaryEncoder()
+        enc.prepare(data)
+
+        enc_data = enc.encode(data)
+        dec_data = enc.decode(enc.encode(test_data))
+
+        self.assertTrue(
+            (
+                enc_data
+                == Tensor(
+                    [
+                        [
+                            1.0,
+                            0.0,
+                        ],  # category 1
```

Comment:
Are comments correct here? I spot a couple odd ones

Reply:
ah must be a holdover will fix